### ISSUE TITLE
All Items Selected flag on Bulk Commands not working correctly

### ISSUE THREAD

**Issue by dehivis24** [Sitefinity/sitefinity-admin-app-extensions#355]
Hello,

There is an issue with the flag that indicates if all content items were selected while executing a custom bulk command. The flag is set to true even though all content items are not selected.

![image](https://user-images.githubusercontent.com/69977848/131758433-b9e2e09f-c991-4380-97fe-70099a6fbd05.png)

How to replicate:
- Select all content items. (At this moment the flag is set to true which is correct)
- Unselect one of the content items and execute the command. (The flag should be set to false but it is still true)

![image](https://user-images.githubusercontent.com/69977848/131758144-4b6284b7-aa98-4adc-8181-8bc97da7fa4e.png)

Thanks.

**Comment by Novkirishki** [comment]
Hi @dehivis24,

Thank you for submitting this. 

The **allItemsSelected** flag is used to mark whether the user has selected all items at some point. It is used to determine whether there are selected items that are not currently loaded in the grid. Deselecting few items would not reset the flag. It will reset when:

- **clear** button is clicked
- more items are loaded
- operation is executed

What exactly is your use case? What data do you need to have when executing an operation? Would adding a collection with deselected items (similar to **selectedItems**)  be sufficient?

**Comment by dehivis24** [comment]
Hi Nikolai,

Thanks for answering so quickly. 

Our use case is the following:
We have a custom command for our Products content type that was added on the grid options menu (it can be executed single or in bulk). When this action is executed a custom DynamicContentOperation is called from that command, this operation requires all the guids of the content items that were selected during the execution of our custom command.

We usually work with a large number of products so the guids that are sent to the DynamicContentOperation can be quite a lot, so we decided that in case a user selects ALL of the products instead of sending all of the guids, we send a boolean flag that indicates that all products were selected and this boolean flag is set according to the allItemsSelected flag provided by your extensions. After some testing, we found the bug described in this post so our command is not working 100% correctly.

I think that a collection with deselectedItems could work for us because we could validate if ALL content items were selected or not by using the allItemsSelected flag and the deselectedItems collection.
